### PR TITLE
add search project id function

### PR DESCRIPTION
**Pull request by letusfly85**
I want to get GitLab repo's projectId below.

https://github.com/letusfly85/drone/blob/master/plugin/remote/gitlab/gitlab.go#L165

### CODE REVIEW

File: deploy_keys.go [hunk 1]
```diff
@@ -47,6 +49,41 @@ Parameters:
     key_id The ID of a key
 
 */
+func (g *Gitlab) SearchProjectId(namespace string, name string) (id int, err error) {
```
Comment (Bugagazavr) [inline deploy_keys.go]: Please, update godoc comments before this method or remove comments

File: deploy_keys.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ const (
 	project_url_deploy_keys = "/projects/:id/keys" // Get list of project deploy keys
 	// PROJECT ID AND KEY ID
 	project_url_deploy_key = "/projects/:id/keys/:key_id" // Get single project deploy key
+
+	project_url_search_ids = "/projects/search/:query"
```
Comment (Bugagazavr) [inline deploy_keys.go]: i think this should be moved to projects.go too

**Comment by letusfly85** [comment]
Thank you.

I moved function to projects.go,
and added comments.

**Comment by letusfly85** [comment]
Thank you!

I changed const from 'project_url_search_ids' to 'projects_search_url' in projects.go.
And I removed 'project_url_search_ids' from deploy_key.go.

**Comment by letusfly85** [comment]
Modified and I write a test code.

Please check.

**Comment by Bugagazavr** [comment]
thanks

**Comment by letusfly85** [comment]
Thank you!

**Pull request merged**